src/jpgimage.cpp

```diff
@@ -75,12 +75,15 @@ bool markerHasLength(byte m) {
   return !markerWithoutLength;
 }
 
-void readSegmentSize(const byte marker, BasicIo& io, std::array<byte, 2>& buf, std::uint16_t& size) {
+std::pair<std::array<byte, 2>, uint16_t> readSegmentSize(const byte marker, BasicIo& io) {
+  std::array<byte, 2> buf{0, 0};  // 2-byte buffer for reading the size.
+  uint16_t size{0};               // Size of the segment, including the 2-byte size field
   if (markerHasLength(marker)) {
     io.readOrThrow(buf.data(), buf.size(), ErrorCode::kerFailedToReadImageData);
     size = getUShort(buf.data(), bigEndian);
     enforce(size >= 2, ErrorCode::kerFailedToReadImageData);
   }
+  return std::make_pair(buf, size);
```

Comment:
should use {} instead of make_pair.

Btw, because of CTAD, std::pair is an alternative to make_pair.